site/gatsby-site/src/components/ReportedIncident.js

```diff
@@ -79,100 +61,109 @@ const ReportedIncident = ({ incident: report }) => {
 
   const isSubmitter = isRole('submitter');
 
-  const updateLinkedReports = useUpdateLinkedReports();
+  const [promoteSubmission] = useMutation(PROMOTE_SUBMISSION, { fetchPolicy: 'network-only' });
 
-  const [getLastIndexes] = useLazyQuery(lastIndexesQuery, { fetchPolicy: 'network-only' });
+  const [updateReport] = useMutation(UPDATE_REPORT);
 
-  const [insertReport] = useMutation(INSERT_REPORT);
-
-  const [deleteSubmission] = useMutation(DELETE_SUBMISSION);
-
-  const [insertIncident] = useMutation(INSERT_INCIDENT);
+  const [updateIncident] = useMutation(UPDATE_INCIDENT);
 
   const [updateSubmission] = useMutation(UPDATE_SUBMISSION);
 
+  const [deleteSubmission] = useMutation(DELETE_SUBMISSION);
+
   const addToast = useToastContext();
 
   const addReport = async () => {
-    const newReport = { ...report };
-
-    delete newReport.mongodb_id;
-
-    delete newReport._id;
-
     const {
       data: {
-        lastIncident: [{ incident_id: lastIncidentId }],
-        lastReport: [{ report_number: lastReportNumber }],
-        refsNumbers,
+        promoteSubmissionToReport: { 0: incident },
```

Comment:
Why is this `{ 0: incident }`? Is this something the Realm function unpacks as an input variable?

Reply:
The resolver can handle promoting a submission to a report that's linked to many incidents,  but at the moment we are only promoting to only, so it's getting that one.